cli/cook/cli.py

```diff
@@ -93,10 +94,18 @@ def run(args, plugins):
             defaults = config_map.get('defaults')
             action_defaults = (defaults.get(action) if defaults else None) or {}
             # load_target_clusters needs to be aware if there are any action-specific cluster defaults
+            # We want to treat cluster, pool and url as a tuple here, so if any are specified on the cli,
+            # we will override the defaults.
+            # Additionally, we must `pop` to remove the disallowed 'cluster' key, regardless of
+            # whether we use the action_defaults or the cli flags
             if action_defaults:
-                # coalesce the defaults with cli flags overriding action config defaults
-                # using `pop` to remove the disallowed 'cluster' key
-                cluster = cluster or action_defaults.pop("cluster", None)
+                action_default_cluster = action_defaults.pop("cluster", None)
+                action_default_url = action_defaults.pop("url", None)
+                if cluster or pool_name or url:
```

Comment:
I added action-default url logic to fix the cli integration tests. The issue is that the cli will always include the local `.cs.json` config as the base of cli invocations, so the default submit cluster was triggering our cluster vs url validation.